packages/lesswrong/components/editor/ReactCKEditor.jsx

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Watchdog } from '@lesswrong/lesswrong-editor'
+
+export default class CKEditor extends React.Component {
+  constructor( props ) {
+    super( props );
+    
+    // After mounting the editor, the variable will contain a reference to the created editor.
+    // @see: https://ckeditor.com/docs/ckeditor5/latest/api/module_core_editor_editor-Editor.html
+    this.editor = null;
+    this.domContainer = React.createRef();
+    this.watchdog = new Watchdog()
+  }
+  
+  // This component should never be updated by React itself.
+  shouldComponentUpdate( nextProps ) {
+    if ( !this.editor ) {
+      return false;
+    }
+    
+    if ( this._shouldUpdateContent( nextProps ) ) {
+      this.editor.setData( nextProps.data );
+    }
+    
+    if ( 'disabled' in nextProps ) {
+      this.editor.isReadOnly = nextProps.disabled;
+    }
+    
+    return false;
+  }
+  
+  // Initialize the editor when the component is mounted.
+  componentDidMount() {
+    this._initializeEditor();
+  }
+  
+  // Destroy the editor before unmouting the component.
+  componentWillUnmount() {
+    this._destroyEditor();
+  }
+  
+  // Render a <div> element which will be replaced by CKEditor.
+  render() {
+    // We need to inject initial data to the container where the editable will be enabled. Using `editor.setData()`
+    // is a bad practice because it initializes the data after every new connection (in case of collaboration usage).
+    // It leads to reset the entire content. See: #68
+    return (
+      <div ref={ this.domContainer } dangerouslySetInnerHTML={ { __html: this.props.data || '' } }></div>
+    );
+  }
+    
+  _initializeEditor() {
+    this.watchdog.setCreator((el, config) => {
+      this.props.editor
+        .create( el , config )
+        .then( editor => {
+          this.editor = editor;
+          
+          if ( 'disabled' in this.props ) {
+            editor.isReadOnly = this.props.disabled;
+          }
+          
+          if ( this.props.onInit ) {
+            this.props.onInit( editor );
+          }
+          
+          const modelDocument = editor.model.document;
+          const viewDocument = editor.editing.view.document;
+          
+          modelDocument.on( 'change:data', event => {
+            /* istanbul ignore else */
```

Comment:
Is this pasted from somewhere?

Reply:
Ah, yes. Let me add the source (it's the component we were using anyways, just copy pasted and slightly modified)